src/DwitTech.AccountService.Core/Services/UserService.cs

```diff
@@ -1,10 +1,99 @@
+using DwitTech.AccountService.Core.Dtos;
+using DwitTech.AccountService.Core.Interfaces;
+using DwitTech.AccountService.Core.Utilities;
 using DwitTech.AccountService.Data.Entities;
 using DwitTech.AccountService.Data.Repository;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
 
 namespace DwitTech.AccountService.Core.Services
 {
     public class UserService : IUserService
     {
+
+        private readonly IUserRepository _userRepository;
+        private readonly IRoleRepository _roleRepository;
+        private readonly ILogger<UserService> _logger;
+        private readonly IActivationService _activationService;
+        private readonly IEmailService _emailService;
+        public UserService(IUserRepository userRepository, IRoleRepository roleRepository, ILogger<UserService> logger, IActivationService activationService, IEmailService emailService)
+        {
+            _userRepository = userRepository;
+            _roleRepository = roleRepository;
+            _logger = logger;
+            _activationService = activationService;
+            _emailService = emailService;
+        }
+
+        private async Task<Data.Entities.Role> GetAssignedRole(UserDto user)
+        {
+            var roles = await _roleRepository.GetRoles();
+            if (user.Roles.ToString() is not null)
+            {
+                var assignedRole = roles.FirstOrDefault(x => x.Id == ((int)user.Roles));
+                if (assignedRole != null)
+                {
+                    return assignedRole;
+                }
+            }
+            return roles.First(x => x.Id == 2);
+
+        }
+
+        private User GetCustomMapper(UserDto user, Data.Entities.Role userIdentifiedRole)
+        {
+            return new User
+            {
+                FirstName = user.FirstName,
+                LastName = user.LastName,
+                Roles = userIdentifiedRole,
+                Country = user.Country,
+                State = user.State,
+                Email = user.Email,
+                PassWord = user.PassWord,
+                AddressLine1 = user.AddressLine1,
+                AddressLine2 = user.AddressLine2,
+                ZipCode = user.ZipCode,
+                PostalCode = user.PostalCode,
+                PhoneNumber = user.PhoneNumber,
+                City = user.City
+            };
+        }
+
+        private UserLogin GenerateLoginCredentials(UserDto user, int id)
+        {
+
+            return new UserLogin
+            {
+                UserName = user.Email,
+                PassWord = StringUtil.HashString(user.PassWord),
+                UserId = id
+            };
+        }
+
+
+        public async Task CreateUser(UserDto user)
+        {
+            try
+            {
+                Data.Entities.Role userRole = await GetAssignedRole(user);
+                var userModel = GetCustomMapper(user, userRole);
+                var emailHtmlTemplate = "EmailTemplate.html";
```

Comment:
I think it'll be better to create a const variable for this activation email template and pass it in here, rather than hardcode it